src/System.Security.Principal.Windows/src/System/Security/Principal/SID.cs

```diff
@@ -792,21 +792,35 @@ public override string ToString()
         {
             if (_sddlForm == null)
             {
-                StringBuilder result = new StringBuilder();
-
                 //
                 // Typecasting of _IdentifierAuthority to a long below is important, since
                 // otherwise you would see this: "S-1-NTAuthority-32-544"
                 //
-
-                result.Append("S-1-").Append((long)_identifierAuthority);
-
-                for (int i = 0; i < SubAuthorityCount; i++)
+                // length of buffer calculation
+                // prefix = "S-1-".Length: 4;
+                // authority: long.MaxValue.ToString("D") : 19;
+                // subauth = MaxSubAuthorities * ( uint.MaxValue.ToString("D").Length + '-'.Length ): 15 * (10+1): 165;
+                // max possible length = 4 + 19 + 165: 188
+                Span<char> result = stackalloc char[188];
+                result[0] = 'S';
+                result[1] = '-';
+                result[2] = '1';
+                result[3] = '-';
+                int written;
+                int length = 4;
+                ((ulong)_identifierAuthority).TryFormat(result.Slice(4), out written);
+                length += written;
+                int[] values = _subAuthorities;
+                for (int index = 0; index < values.Length; index++)
                 {
-                    result.Append('-').Append((uint)(_subAuthorities[i]));
+                    uint value = (uint)values[index];
```

Comment:
Caching the value in a local here is not helping clarity nor performance. I would just call `TryFormat` directly on the array element below:

`((uint)values[index]).TryFormat(result.Slice(length), out written);`

Reply:
I think it's clearer as a local, but since I've done it the unclear way with the authorityIdentifier It clearly isn't that important to me. Changed.